### DISCUSSION TITLE
Stencil elements are smaller than they should be

### DISCUSSION THREAD

**Discussion by frnora** [clientIO/joint#2105] [Q&A] [answered]
### Introduction

Hi there,

This is just out of curiosity because I know I can achieve this without grid layout.

Let's say I am trying to create a stencil with 2 items. 1 row, 2 columns. Let's say I want every item to take up max 30x30 in the stencil, so I will set the stencil width to 60 and height to 30.

Now this all works but as soon as I add the `layout` config, it shrinks the stencil items. This is the code of the stencil:
```js
const stencil = new ui.Stencil({
      graph: this.#graph,
      paper: this.#paperScroller,
      width: 60,
      height: 30,
      layout: {
           columnWidth: 30,
           columns: 2,
           rowHeight: 30,
           resizeToFit: true // ? makes no difference
      }
});
```
The CSS is not really interesting, it has a `position`, `width` and `height` (same values as the stencil).

I have modified one of your demos to show the outcome: https://codepen.io/frnora/pen/ExezZWY
You can see the stencil elements are max 10px in width & height. There seems to be a 20px margin-left and top around them. I would expect them to fill 30px (width/height, whichever possible).

The only thing I could find that seems to work is to set `dx` and `dy`. Both of them are deprecated tho. I can't seem to achieve it in any other way. What would be the correct approach?

Thank you!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
The `dx` and  `dy` default values are `10`. You have to set them to `0` explicitly.

```js
const stencil = new ui.Stencil({
      graph: this.#graph,
      paper: this.#paperScroller,
      width: 60,
      height: 30,
      layout: {
           columnWidth: 30,
           columns: 2,
           rowHeight: 30,
           resizeToFit: true,
           // reset the undesired defaults
           dx: 0,
           dy: 0,
      }
});
```

The `dx` and `dy` are indeed deprecated and will be removed in `v4`.
